src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/CreateComponentCodeActionResolver.cs

```diff
@@ -78,23 +77,15 @@ private static void TryAddNamespaceDirective(RazorCodeDocument codeDocument, Uri
         var namespaceDirective = syntaxTree.Root.DescendantNodes()
             .Where(n => n.Kind == SyntaxKind.RazorDirective)
             .Cast<RazorDirectiveSyntax>()
-            .Where(n => n.DirectiveDescriptor == NamespaceDirective.Directive)
-            .FirstOrDefault();
+            .FirstOrDefault(static n => n.DirectiveDescriptor == NamespaceDirective.Directive);
 
         if (namespaceDirective != null)
         {
             var documentIdentifier = new OptionalVersionedTextDocumentIdentifier { Uri = newComponentUri };
             documentChanges.Add(new TextDocumentEdit
             {
                 TextDocument = documentIdentifier,
-                Edits =
-                [
-                    new TextEdit()
-                    {
-                        NewText = namespaceDirective.GetContent(),
-                        Range = new Range{ Start = new Position(0, 0), End = new Position(0, 0) },
-                    }
-                ]
+                Edits = [VsLspFactory.CreateTextEdit(VsLspFactory.EmptyRange, namespaceDirective.GetContent())]
```

Comment:
Good feedback, even if it's bike-shedding! I took `Empty` from the `Point` and `Rectangle` APIs, which are very similar in concept to `Position` and `Range`. `Rectangle` is made up of two `Points` in the top-left and bottom-right corners, and `Range` has the same relationship with `Position`. However, I totally get your point (ha!) about collapsed vs. emptiness. I'll update the names.

Reply:
And FWIW, I _really_ do appreciate this feedback and your suggestions. `Rectangle.IsEmpty` has always been an API stumbling block for many because it means `x == 0 && y == 0 && width == 0 && height == 0` rather than `width == 0 && height == 0`.

Reply:
I ended up changing "Empty" to "Default" and "Collapsed" to "ZeroWidth". It all feels better.

Reply:
> Rectangle.IsEmpty has always been an API stumbling block for many because it means x == 0 && y == 0 && width == 0 && height == 0 rather than width == 0 && height == 0.

This would annoy me for sure, and I would definitely argue there is no such thing as an "empty point" too. Good thing I'm not on API review :P